documentation/src/test/java/org/hibernate/validator/referenceguide/chapter12/nodenameprovider/JacksonPropertyNodeNameProvider.java

```diff
@@ -4,42 +4,85 @@
  */
 package org.hibernate.validator.referenceguide.chapter12.nodenameprovider;
 
-//tag::include[]
 import org.hibernate.validator.spi.nodenameprovider.JavaBeanProperty;
 import org.hibernate.validator.spi.nodenameprovider.Property;
 import org.hibernate.validator.spi.nodenameprovider.PropertyNodeNameProvider;
 
-import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.databind.BeanProperty;
 import com.fasterxml.jackson.databind.JavaType;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.introspect.BeanPropertyDefinition;
+import com.fasterxml.jackson.databind.introspect.AnnotatedMethod;
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonFormatVisitorWrapper;
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonObjectFormatVisitor;
+
+//tag::include[]
 
 public class JacksonPropertyNodeNameProvider implements PropertyNodeNameProvider {
-	private final ObjectMapper objectMapper = new ObjectMapper();
-
-	@Override
-	public String getName(Property property) {
-		if ( property instanceof JavaBeanProperty ) {
-			return getJavaBeanPropertyName( (JavaBeanProperty) property );
-		}
-
-		return getDefaultName( property );
-	}
-
-	private String getJavaBeanPropertyName(JavaBeanProperty property) {
-		JavaType type = objectMapper.constructType( property.getDeclaringClass() );
-		BeanDescription desc = objectMapper.getSerializationConfig().introspect( type );
-
-		return desc.findProperties()
-				.stream()
-				.filter( prop -> prop.getInternalName().equals( property.getName() ) )
-				.map( BeanPropertyDefinition::getName )
-				.findFirst()
-				.orElse( property.getName() );
-	}
-
-	private String getDefaultName(Property property) {
-		return property.getName();
-	}
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public String getName(Property property) {
+        if (property instanceof JavaBeanProperty javaBeanProperty) {
+            var visitor = new JsonPropertyNameGetter(property.getName());
```

Comment:
Let's go with:
```suggestion
            var visitor = new JsonPropertyNameGetter(javaBeanProperty.getMemberName());
```
This way we'll have the field/method name and wouldn't need to "guess" how the getter name is actually formatted. 
I've added it here: https://github.com/hibernate/hibernate-validator/pull/1779/changes